pkg/generic/ensure_release_label_test.go

```diff
@@ -0,0 +1,221 @@
+			meta:         newObjectWithRelease(to.StringPtr("cd456"), nil),
+			patch:        nil,
+			errorMatcher: errors.IsInvalidOperationError,
+		},
+		{
+			name:         "case 3: release wasn't set, cluster CR found but without a release label",
+			meta:         newObjectWithRelease(to.StringPtr("ef789"), nil),
+			patch:        nil,
+			errorMatcher: errors.IsInvalidOperationError,
+		},
+		{
+			name: "case 4: release wasn't set, cluster CR found, release label present",
+			meta: newObjectWithRelease(to.StringPtr("ab123"), nil),
+			patch: &mutator.PatchOperation{
+				Operation: "add",
+				Path:      "/metadata/labels/release.giantswarm.io~1version",
+				Value:     "13.0.0",
+			},
+			errorMatcher: nil,
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			t.Log(tc.name)
+
+			ctx := context.Background()
+			fakeK8sClient := unittest.FakeK8sClient()
+			ctrlClient := fakeK8sClient.CtrlClient()
+
+			ab123 := &v1alpha3.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ab123",
+					Namespace: "default",
+					Labels: map[string]string{
+						"release.giantswarm.io/version": "13.0.0",
+					},
+				},
+			}
+			err := ctrlClient.Create(ctx, ab123)
+			if err != nil {
+				t.Fatal(err)
+			}
+			ef789 := &v1alpha3.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ef789",
+					Namespace: "default",
+				},
+			}
+			err = ctrlClient.Create(ctx, ef789)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			patch, err := EnsureReleaseVersionLabel(ctx, ctrlClient, tc.meta)
+
+			switch {
+			case err == nil && tc.errorMatcher == nil:
+				// correct; carry on
+			case err != nil && tc.errorMatcher == nil:
+				t.Fatalf("error == %#v, want nil", err)
+			case err == nil && tc.errorMatcher != nil:
+				t.Fatalf("error == nil, want non-nil")
+			case !tc.errorMatcher(err):
+				t.Fatalf("error == %#v, want matching", err)
+			}
+
+			// Check if the validation result is the expected one.
+			if !reflect.DeepEqual(tc.patch, patch) {
+				t.Fatalf("Patch mismatch: expected %v, got %v", tc.patch, patch)
+			}
+		})
+	}
+}
+
+func newObjectWithRelease(clusterID *string, release *string) metav1.Object {
+	obj := &GenericObject{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Unknown",
+			APIVersion: "unknown.generic.example/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ab123",
+			Namespace: "default",
+			Labels: map[string]string{
+				"azure-operator.giantswarm.io/version": "5.0.0",
+				"cluster.x-k8s.io/cluster-name":        "ab123",
+				"cluster.x-k8s.io/control-plane":       "true",
+				"giantswarm.io/machine-pool":           "ab123",
+			},
+		},
+	}
+
+	if clusterID != nil {
+		obj.Labels[label.Cluster] = *clusterID
+	}
+
+	if release != nil {
+		obj.Labels[label.ReleaseVersion] = *release
+	}
+
+	return obj
+}
+
+//func Test_WhenCreatingClusterWithExistingOrganizationThenValidationSucceeds(t *testing.T) {
```

Comment:
If this is not needed, please delete it so it doesn't confuse anyone in the future :slightly_smiling_face: